packages/core/helper-plugin/src/content-manager/CMEditViewDataManager.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+
+import { MessageDescriptor } from 'react-intl';
+
+/**
+ * TODO: All these types could be accurately inferred from
+ * their usage in the content-manager when we move it back.
+ */
+interface CMEditViewDataManagerContext {
+  addComponentToDynamicZone?: (
+    keys: string[],
+    componentLayoutData: Record<string, unknown>,
+    allComponents: Record<string, unknown>,
+    shouldCheckErrors?: boolean,
+    position?: number
+  ) => void;
+  addNonRepeatableComponentToField?: (
+    keys: string[],
+    componentLayoutData: Record<string, unknown>,
+    allComponents: Record<string, unknown>
+  ) => void;
+  addRepeatableComponentToField?: (
+    keys: string[],
+    componentLayoutData: Record<string, unknown>,
+    allComponents: Record<string, unknown>,
+    shouldCheckErrors?: boolean,
+    position?: number
+  ) => void;
+  allLayoutData: {
+    // TODO: this should be a schema?
+    components: Record<string, unknown>;
+    // TODO: this should be a schema?
+    contentType: Record<string, unknown>;
+  };
+  createActionAllowedFields: string[];
+  formErrors: Record<string, MessageDescriptor>;
+  hasDraftAndPublish: boolean;
+  // TODO: define the tree, can it be inferred?
+  initialData: Record<string, unknown>;
+  isCreatingEntry: boolean;
+  isSingleType: boolean;
+  // TODO: this should be a schema?
+  layout: Record<string, unknown>;
+  // TODO: define the tree, can it be inferred?
+  modifiedData: Record<string, unknown>;
+  moveComponentDown?: (dynamicZoneName: string, currentIndex: number) => void;
+  moveComponentField?: (payload: { name: string; newIndex: number; currentIndex: number }) => void;
+  moveComponentUp?: (dynamicZoneName: string, currentIndex: number) => void;
+  onChange?: (
+    payload: { target: { name: string; type: string; value: string } },
+    shouldSetInitialValue?: boolean
+  ) => void;
```

Comment:
I'll need this too for the GenericInput could we move it to the types file? I'll share what I had come up with let me know what you think.

```suggestion
  onChange?: (
    payload: { target: { name: string; type?: Attribute.Kind; value: string | number | boolean | Attribute.Media | null } },
    shouldSetInitialValue?: boolean
  ) => void;
```

In my case type is not always passed..which confuses me...

Reply:
I would move it when you're doing GenericInput to see how it works with that and see if you actually need it shared as opposed to anticipating, just incase for whatever reason, you don't need it and it's automatically inferred.